tests/unit/backend/survey.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { expect, jest } from "@jest/globals"
+import { SurveyType } from "../../../lib/enums/survey"
+
+describe("Survey methods", () => {
+  let mockFollowup
+
+  beforeEach(() => {
+    mockFollowup = {
+      surveys: [],
+      save: jest.fn(),
+      addSurveyIfMissing: jest.fn(),
+      updateSurvey: jest.fn(),
+    }
+  })
+
+  describe("addSurveyIfMissing", () => {
+    it("should add survey if not exists", async () => {
+      const expectedSurvey = {
+        type: SurveyType.TrackClickOnSimulationUsefulnessEmail,
+      }
+
+      mockFollowup.addSurveyIfMissing.mockResolvedValue(expectedSurvey)
+
+      const result = await mockFollowup.addSurveyIfMissing(
+        SurveyType.TrackClickOnSimulationUsefulnessEmail
+      )
+
+      expect(mockFollowup.addSurveyIfMissing).toHaveBeenCalledWith(
+        SurveyType.TrackClickOnSimulationUsefulnessEmail
+      )
+      expect(result).toEqual(expectedSurvey)
+    })
+  })
+
+  describe("updateSurvey", () => {
+    it("should update survey with answers", async () => {
+      const answers = [
+        {
+          id: "wasUseful",
+          value: true,
+        },
+      ]
+
+      mockFollowup.updateSurvey.mockResolvedValue(undefined)
+
+      await mockFollowup.updateSurvey(
+        SurveyType.TrackClickOnSimulationUsefulnessEmail,
+        answers
+      )
+
+      expect(mockFollowup.updateSurvey).toHaveBeenCalledWith(
+        SurveyType.TrackClickOnSimulationUsefulnessEmail,
+        answers
+      )
+    })
+  })
```

Comment:
En associant la vraie méthode updateSurvey dans le mock

```js
import FollowupModel from "../../../backend/models/followup.js"
...
mockFollowup = {
      surveys: [],
      save: jest.fn(),
      addSurveyIfMissing: jest.fn(),
      updateSurvey: FollowupModel.prototype.updateSurvey,
    }
```

On peut alors tester le comportement de updateSurvey (sans l'appel à la base)

```suggestion
  describe("updateSurvey", () => {
    it("devrait mettre à jour les réponses d'une enquête existante", async () => {
      const existingSurvey: Survey = {
        _id: "test-id",
        type: SurveyType.TrackClickOnSimulationUsefulnessEmail,
        answers: [],
        touchedAts: [],
      }

      mockFollowup.surveys = [existingSurvey]
      const newAnswers = [{ id: "wasUseful", value: true }]

      await mockFollowup.updateSurvey(
        SurveyType.TrackClickOnSimulationUsefulnessEmail,
        newAnswers
      )

      expect(mockFollowup.surveys[0].answers).toEqual(newAnswers)
      expect(mockFollowup.surveys[0].touchedAts).toHaveLength(1)
    })
  })
```

Reply:
C'est bien intégré, j'ai juste conservé mon title en anglais.

Reply:
oui, erreur de ma part à ce sujet !